Update Poco to 1.9.3.

akuzm opened the pull request:
This is required for MongoDB URI support.

Category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry:
Update Poco to 1.9.3 in preparation for MongoDB URI support.

alesapin commented:
Clone error:
```
No submodule mapping found in .gitmodules for path 'gradle'
Failed to recurse into submodule path 'contrib/poco'
```

akuzm committed:
Update Poco to 1.9.3

akuzm committed:
Register Poco ODBC Connector on clickhouse-odbc-bridge startup

akuzm commented:
Should update to 1.9.4: https://pocoproject.org/blog/?p=1099

akuzm committed:
More logs to catch the ODBC bridge shutdown error.

Also increase a potentially flappy timeout in another test.

akuzm committed:
More logs + build fix.

alexey-milovidov committed:
Update ShellCommand.cpp

alexey-milovidov committed:
Update ShellCommand.h

alexey-milovidov commented on libs/libdaemon/src/BaseDaemon.cpp:
```diff
@@ -148,6 +148,7 @@ class SignalListener : public Poco::Runnable
         {
             int sig = 0;
             DB::readBinary(sig, in);
+            LOG_TRACE(log, "Received signal " << sig);
```
1. strsignal.
2. We already print it somewhere.

akuzm committed:
Use strsignal.

akuzm committed:
Remove trailing whitespace.

filimonov commented:
Can you also look on that while we are here?
https://github.com/ClickHouse/ClickHouse/issues/7736#issuecomment-552922270

akuzm commented:
> Can you also look on that while we are here?
> [#7736 (comment)](https://github.com/ClickHouse/ClickHouse/issues/7736#issuecomment-552922270)

Not until I finish this update, and my perspective on this is quite grim, given the Mystery Integration Test Failure I'm seeing in CI but not locally.

akuzm committed:
test

akuzm committed:
Reset signal mask before calling exec().

It might be non-empty and the child process might not expect this. For
example, in one case we started clickhouse-odbc-bridge with blocked
SIGTERM and then it didn't shutdown properly along with the server.

akuzm committed:
Straighten the ODBC integration test somewhat.

akuzm committed:
Remove global scope from sigprocmask etc.

These functions may be macros on some platforms e.g. Mac OS

akuzm committed:
Check order values in clickhouse-test

akuzm committed:
Use the appropriate Int64 variant.

Our Int64 is int64_t, and Poco's is long long, which are distinct for
the purposes of template specialization on MacOS.

akuzm committed:
Doesn't look like I'll ever complete this.

alexey-milovidov commented:
Tsan report is relevant and it looks familiar to me.
Did you integrate all the patches?

alexey-milovidov commented:
It looks almost ready, let's finish.

alexey-milovidov commented:
`00834_kill_mutation` - flacky test.

`test_dictionaries_update_and_reload/test.py::test_reload_after_loading` - it may be attributed to https, let's check carefully.

akuzm committed:
Relax a test due to Poco regression.

Poco 1.9.3 has one second precision for mtime, because of this we can
miss an update of a dictionary file if it happens in the same second
we've read the file.

Should probably be fixed by switching to std::filesystem.

Also add some logs because finding out what's going on with dictionary
reloading is borderline impossible.

akuzm commented:
Split build failed because of CI error -- works locally.

alexey-milovidov commented:
Split build smoke test: #10820
